### PR TITLE
Clean way to disable the pcov PHP Extension

### PR DESCRIPTION
**Pull request by sylfabre**
# Description
Use phpdismod to disable th pcov PHP extension

#### Related issue:
https://github.com/actions/virtual-environments/issues/3470

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Comment by miketimofeev** [comment]
@shivammathur could you please also take a look at this PR?

**Comment by shivammathur** [comment]
@sylfabre 
Using `phpdismod` would create a breaking change in `shivammathur/setup-php` action. That is why I choose to edit the module ini file, which would not.

Anyone using `pcov` can add `extension=pcov.so` to a ini file. This is consistent with PHP documentation for loading an extension.

But I understand that `phpenmod` and `phpdismod` by `ondrej` are conventionally used in ubuntu, so if it is too much trouble for Azure users, I'm ok with using `phpdismod` and will make the necessary changes to setup-php. 

So, if using you are using `phpdismod`, I would recommend adding a line in the docs for ubuntu below PHP version table mentioning that `pcov` is disabled and can be enabled by using `phpenmod`.

**Comment by sylfabre** [comment]
@shivammathur Thank you for your feedback and explanation.

The files in the `mods-available` folder actually are like templates to be copied/symlinked to `[context]/conf.d` folder of each possible context (apache2, fpm, cli, ...) for which an extension could be enabled. It works the same way as with Apache if you're familiar with it.

So deleting these files' content from `mods-available` is a bad practice.

You can also simply delete the `pcov.ini` file from `conf.d` folders if you want to disable it without the use of an external tool like `phpenmod`/`phpdismod`.

**Comment by sylfabre** [comment]
@shivammathur And you could also challenge the default choice of xdebug over pcov.

In my opinion, pcov is going to be the new standard for code coverage as it runs faster and consumes less memory that xdebug (https://medium.com/@nicocabot/speed-up-phpunit-code-coverage-analysis-4e35345b3dad).

That's why we are using pcov at my company.

But in the end, this is a product choice that belongs to you guys!

**Comment by shivammathur** [comment]
@sylfabre 
They are few, but I know projects testing with php-fpm and other sapi as well. So the set of extensions should be consistent. That is why I changed the main module ini files instead of symlinks in each sapi. I'm ok with changing it to use `phpdismod` as that handles all sapi.

Ideally both pcov and Xdebug should be disabled as they add overhead to PHP and can be enabled if needed in the workflow. But changing this either way would be a breaking change.

**Comment by sylfabre** [comment]
@shivammathur Ok I understand why you changed it that way.

What are the next steps please?

**Comment by shivammathur** [comment]
Documenting that PCOV extension is disabled, with the steps to enable it using `phpenmod` would be great. This can be added below the PHP version table similar to how [MySQL service is documented](https://github.com/actions/virtual-environments/blob/f29cc02bef4dcd5d5132d59c1287cbe5a531278f/images/linux/Ubuntu2004-README.md#mysql).

@miketimofeev What are your thoughts on this.

**Comment by sylfabre** [comment]
@shivammathur I've updated the doc as per your request

File: images/linux/Ubuntu2004-README.md [hunk 1]
```diff
@@ -128,6 +128,12 @@
 | Composer | 2.0.13       |
 | PHPUnit  | 8.5.15       |
 
+Both xdebug and pcov are installed but only xdebug is enabled. Run these commands to switch to pcov:
```
Comment (shivammathur) [inline images/linux/Ubuntu2004-README.md:131]: Docs are generated dynamically. 
Please use the scripts here to update the docs.
https://github.com/actions/virtual-environments/tree/main/images/linux/scripts/SoftwareReport

Comment (sylfabre) [reply images/linux/Ubuntu2004-README.md:131]: @shivammathur I don't know these scripts language so I don't know how to do it.

Can you please do it yourself?

Comment (shivammathur) [reply images/linux/Ubuntu2004-README.md:131]: @sylfabre
Done, 
Remove the changes in the README and add this commit https://github.com/shivammathur/virtual-environments/commit/fa3fcb9fa81708a9241d45593901f0686b6fa4b7 to your PR.

Comment (sylfabre) [reply images/linux/Ubuntu2004-README.md]: Done

Comment (shivammathur) [reply images/linux/Ubuntu2004-README.md]: @sylfabre Please also update the `images/linux/scripts/SoftwareReport/SoftwareReport.Generator.ps1` file to call `Build-PHPSection`. Refer to the second diff in the commit https://github.com/shivammathur/virtual-environments/commit/fa3fcb9fa81708a9241d45593901f0686b6fa4b7

Comment (sylfabre) [reply images/linux/Ubuntu2004-README.md]: @shivammathur Done, sorry for this oversight 🙈

**Review by shivammathur — DISMISSED** [review]

**Comment by AlenaSviridenko** [comment]
Hi @sylfabre, 
we do not mind making code changes, but updating documentation with instructions how to switch different extensions in PHP is a bit out of scope of our README purpose. 
Could you please remove documentation updates, then I will run test CI to validate your changes.

**Comment by sylfabre** [comment]
@AlenaSviridenko I did it as per the request of your colleague https://github.com/actions/virtual-environments/pull/3472#issuecomment-850387764

Let me know if you want the doc or not

**Comment by shivammathur** [comment]
@AlenaSviridenko I would suggest keeping the documentation to avoid further issues regarding this.

**Comment by AlenaSviridenko** [comment]
We'd prefer to avoid adding any tool specific instructions to our readme. I understand that we have some for MySQL for example, but we decided that adding instructions for tools will complexify our documentation. 
I think, we can have a part of the note like "Both Xdebug and PCOV extensions are installed, but only Xdebug is enabled." without instruction how to switch them.

**Comment by shivammathur** [comment]
@AlenaSviridenko Fair enough

**Comment by miketimofeev** [comment]
@sylfabre could you please pull the latest main to your branch and make a change that @AlenaSviridenko suggested in https://github.com/actions/virtual-environments/pull/3472#issuecomment-853698906?

**Comment by sylfabre** [comment]
@miketimofeev Done

**Review by miketimofeev — APPROVED** [review]

**Review by AlenaSviridenko — APPROVED** [review]

**Pull request merged**